### PR TITLE
CDAP-6152 Metrics for logs emitted by programs

### PR DESCRIPTION
**Pull request by gokulavasan**
- Add tags for logs from programs so that metrics for different log levels at program-run-instance granularity is captured

JIRA : https://issues.cask.co/browse/CDAP-6152

Build : http://builds.cask.co/browse/CDAP-DUT4211

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java [hunk 1]
```diff
@@ -49,11 +53,15 @@
...
       .put(FlowletLoggingContext.TAG_FLOWLET_ID, Constants.Metrics.Tag.FLOWLET)
+      .put(FlowletLoggingContext.TAG_FLOW_ID, Constants.Metrics.Tag.FLOW)
       .put(WorkflowLoggingContext.TAG_WORKFLOW_ID, Constants.Metrics.Tag.WORKFLOW)
       .put(MapReduceLoggingContext.TAG_MAP_REDUCE_JOB_ID, Constants.Metrics.Tag.MAPREDUCE)
       .put(SparkLoggingContext.TAG_SPARK_JOB_ID, Constants.Metrics.Tag.SPARK)
-      .put(UserServiceLoggingContext.TAG_USERSERVICE_ID, Constants.Metrics.Tag.HANDLER)
+      .put(UserServiceLoggingContext.TAG_USERSERVICE_ID, Constants.Metrics.Tag.SERVICE)
+      .put(UserServiceLoggingContext.TAG_HANDLER_ID, Constants.Metrics.Tag.HANDLER)
       .put(WorkerLoggingContext.TAG_WORKER_ID, Constants.Metrics.Tag.WORKER)
+      .put(ApplicationLoggingContext.TAG_RUNID_ID, Constants.Metrics.Tag.RUN_ID)
```
Comment (awholegunch) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java]: RUNID_ID? Why not just RUN_ID?

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java [hunk 2]
```diff
@@ -49,11 +53,15 @@
   private static final Map<String, String> LOG_TAG_TO_METRICS_TAG_MAP =
     ImmutableMap.<String, String>builder()
       .put(FlowletLoggingContext.TAG_FLOWLET_ID, Constants.Metrics.Tag.FLOWLET)
+      .put(FlowletLoggingContext.TAG_FLOW_ID, Constants.Metrics.Tag.FLOW)
       .put(WorkflowLoggingContext.TAG_WORKFLOW_ID, Constants.Metrics.Tag.WORKFLOW)
       .put(MapReduceLoggingContext.TAG_MAP_REDUCE_JOB_ID, Constants.Metrics.Tag.MAPREDUCE)
       .put(SparkLoggingContext.TAG_SPARK_JOB_ID, Constants.Metrics.Tag.SPARK)
-      .put(UserServiceLoggingContext.TAG_USERSERVICE_ID, Constants.Metrics.Tag.HANDLER)
+      .put(UserServiceLoggingContext.TAG_USERSERVICE_ID, Constants.Metrics.Tag.SERVICE)
```
Comment (awholegunch) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java]: TAG_USERSERVICE_ID: maybe instead TAG_USER_SERVICE_ID, as that would be more consistent.

**Comment by gokulavasan** [comment]
@poornachandra Please take a look. Thanks!

**Comment by poornachandra** [comment]
LGTM

**Comment by gokulavasan** [comment]
Build passed. Merging.

**Pull request merged**